src/mission_start.cpp

```diff
@@ -1627,3 +1627,92 @@ void mission_start::ranch_bartender_4(mission *miss)
 void mission_start::place_book( mission *)
 {
 }
+
+const tripoint reveal_destination ( const std::string& type, bool random )
+{
+    const tripoint your_pos = g->u.global_omt_location();
+    const tripoint center_pos = random ?
+    overmap_buffer.find_random( your_pos, type, rng( 40, 80 ), false ) :
+    overmap_buffer.find_closest( your_pos, type, rng ( 40, 80 ), false );
+
+    if( center_pos != overmap::invalid_tripoint ) {
+        overmap_buffer.reveal( center_pos, 2 );
+        return center_pos;
+    }
+
+    return overmap::invalid_tripoint;
+}
+
+void reveal_route( mission *miss, const tripoint destination )
+{
+    const tripoint source = g->u.global_omt_location();
+
+    const tripoint source_road = overmap_buffer.find_closest( source, "road", 3, false );
+    const tripoint dest_road = overmap_buffer.find_closest( destination, "road", 3, false );
+
+    if( overmap_buffer.reveal_route( source_road, dest_road ) )
+        add_msg( _( "%s marks as well the road that leads to it..." ), g->find_npc( miss->get_npc_id() )->name.c_str() );
+}
+
+void mission_start::reveal_weather_station( mission *miss )
+{
+    npc *p = g->find_npc( miss->get_npc_id() );
+
+    const tripoint destination = reveal_destination( "station_radio", true );
+    if( destination != overmap::invalid_tripoint )
+    {
+        add_msg( _( "%s have marked the only broadcasting station known to them on your map" ), p->name.c_str() );
```

Comment:
Here and below: `find_npc()` can return `nullptr`. I'm aware that the code was copied from the existing, but the possibility exists.